### PR TITLE
[tinyxml2] Update tinyxml2 fuzzer

### PR DESCRIPTION
**Pull request by gnbon**
Update tinyxml2 fuzzer by myself. Looking forward to approval.

### CODE REVIEW

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
We shouldn't change the name of the existing fuzzer as this will cause issues in the oss-fuzz backend.

I can see the existing harness has `#if defined( _MSC_VER ) || defined (WIN32)` --> will this, however, be true in any OSS-Fuzz build environment?

will await maintainer approval for this.

**Comment by gnbon** [comment]
@DavidKorczynski I made the changes based on your feedback. How do these changes look?

File: projects/tinyxml2/xmltest2.cpp
Comment (DavidKorczynski) [inline projects/tinyxml2/xmltest2.cpp:28]: does this path always exist? `/dev/shm`?

**Comment by DavidKorczynski** [comment]
> @DavidKorczynski I made the changes based on your feedback. How do these changes look?

it looks fine as such but will await maintainer feedback.

File: projects/tinyxml2/xmltest2.cpp
Comment (gnbon) [reply projects/tinyxml2/xmltest2.cpp:28]: Nope. Would it be better to modify it like this?
https://github.com/google/oss-fuzz/blob/7dc680836c9d99a5e1839ca07f60874654727bc8/projects/binutils/fuzz_addr2line.c#L25-L26

**Comment by gnbon** [comment]
@DavidKorczynski The maintainer is not responding. Is a maintainer approval essential? Or, is there any other way for me to obtain permission?

**Comment by leethomason** [comment]
Approved - and thanks for looking into this

**Comment by gnbon** [comment]
@DavidKorczynski Could you accept this PR?

**Review by DavidKorczynski — APPROVED** [review]

**Pull request merged**
